From 972f114a9b8ca4b7e188ff49ab1f8740df5c9f8e Mon Sep 17 00:00:00 2001
From: G Kranthi <gudishax.kranthikumar@intel.com>
Date: Tue, 8 Mar 2016 01:17:46 +0530
Subject: [PATCH 1965/2508] ASoC: Intel: Skylake: Add low latency dai link

commit dfb4a6a54065fea8aaed84dcf3d84d00d8a97acd from
https://github.com/01org/linux-apollolake-i

Low latency feature will run audio stream with
low period and buffer sizes, which gives better
latency performance. This patch will add low latency
dai link support to morganfield machine driver.

Change-Id: Id3f10411007061e7e956f6a3e2d268d067c0e7ed
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-13417
Signed-off-by: G Kranthi <gudishax.kranthikumar@intel.com>
Reviewed-on: https://android.intel.com:443/478487
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/9068
Reviewed-by: Babu, Ramesh <ramesh.babu@intel.com>
Tested-by: Babu, Ramesh <ramesh.babu@intel.com>
---
 sound/soc/intel/boards/mrgfld_dpcm_florida.c |   40 ++++++++++++++++++++++++++
 1 files changed, 40 insertions(+), 0 deletions(-)

diff --git a/sound/soc/intel/boards/mrgfld_dpcm_florida.c b/sound/soc/intel/boards/mrgfld_dpcm_florida.c
index e5be3af..77eeff5 100644
--- a/sound/soc/intel/boards/mrgfld_dpcm_florida.c
+++ b/sound/soc/intel/boards/mrgfld_dpcm_florida.c
@@ -636,6 +636,32 @@ struct snd_soc_dai_link mrgfld_florida_msic_dailink[] = {
 		.init = NULL,
 		.nonatomic = 1,
 	},
+	{
+                .name = "Bxtn Audio Reference cap",
+                .stream_name = "refcap",
+                .cpu_dai_name = "Reference Pin",
+                .codec_name = "snd-soc-dummy",
+                .codec_dai_name = "snd-soc-dummy-dai",
+                .platform_name = "0000:00:0e.0",
+                .init = NULL,
+                .ignore_suspend = 1,
+                .nonatomic = 1,
+                .dynamic = 1,
+                .dpcm_capture = 1,
+        },
+	{
+		.name = "Bxtn LL Audio Port",
+		.stream_name = "Low Latency Audio",
+		.cpu_dai_name = "LowLatency Pin",
+		.codec_name = "snd-soc-dummy",
+		.codec_dai_name = "snd-soc-dummy-dai",
+		.platform_name = "0000:00:0e.0",
+		.dpcm_playback = 1,
+		.ignore_suspend = 1,
+		.nonatomic = 1,
+		.dynamic = 1,
+		.ops = &mrgfld_florida_ops,
+	},
 
 		/* CODEC<->CODEC link */
 	{
@@ -829,6 +855,20 @@ struct snd_soc_dai_link mrgfld_wm8998_msic_dailink[] = {
 		.nonatomic = 1,
 	},
 
+	{
+		.name = "Bxtn LL Audio Port",
+		.stream_name = "Low Latency Audio",
+		.cpu_dai_name = "LowLatency Pin",
+		.codec_name = "snd-soc-dummy",
+		.codec_dai_name = "snd-soc-dummy-dai",
+		.platform_name = "0000:00:0e.0",
+		.dpcm_playback = 1,
+		.ignore_suspend = 1,
+		.nonatomic = 1,
+		.dynamic = 1,
+		.ops = &mrgfld_florida_ops,
+	},
+
 		/* CODEC<->CODEC link */
 	{
 		.name = "Bxtn Codec-Loop Port",
-- 
1.7.5.4

